tests/check/complexity_three.go

```diff
@@ -661,4 +661,75 @@ var complexityThreeTestingModelTest = []*stage{
 			},
 		},
 	},
+	{
+		Name: "cycle_nested",
+		Tuples: []*openfgav1.TupleKey{
+			{Object: "complexity3:complexity3_cycle_nested_1", Relation: "cycle_nested", User: "ttus:complexity3_cycle_nested_1#tuple_cycle3"},
+			{Object: "ttus:complexity3_cycle_nested_1", Relation: "userset_parent", User: "usersets-user:complexity3_cycle_nested_1"},
+			{Object: "usersets-user:complexity3_cycle_nested_1", Relation: "tuple_cycle3", User: "directs-user:complexity3_cycle_nested_1#compute_tuple_cycle3"},
+			{Object: "directs-user:complexity3_cycle_nested_1", Relation: "tuple_cycle3", User: "complexity3:complexity3_cycle_nested_1#cycle_nested"},
+			{Object: "directs-user:complexity3_cycle_nested_1", Relation: "tuple_cycle3", User: "user:complexity3_cycle_nested_1"},
```

Comment:
Ironically, it's not these set of tuples that produce a cycle, it's the `missing user` tuples below (test cases `invalid_user` and `missing_user`. You can confirm this by adding a print here https://github.com/openfga/openfga/blob/dc6f9d2e88f8d79d22d37833fef9248445264141/internal/graph/check.go#L605 and see which tests print it.

It's because the tuple in line 671 is "breaking" out of the cycle.

<img width="1117" alt="image" src="https://github.com/user-attachments/assets/c3dd561d-9fed-4df1-bb95-2916204497a8">

Reply:
This is all to say, maybe the IDs used need to be changed so that it's clearer which tuples create the cycle

Reply:
Updated test name and variable